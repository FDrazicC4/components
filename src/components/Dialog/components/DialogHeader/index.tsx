import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


function DialogHeader(props: Props): ReactElement {
    const { className, ...otherProps } = props;
    const classes = clsx(
        'brew-Dialog__header',
        className,
    );

    return (
        <div className={classes} {...otherProps} />
    );
}

export default DialogHeader;