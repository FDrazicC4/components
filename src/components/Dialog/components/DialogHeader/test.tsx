import React from 'react';
import renderer from 'react-test-renderer';
import DialogHeader from './index';
import { shallow } from 'enzyme';

describe('DialogHeader', () => {
    it('matches the current snapshot', () => {
        const dialogHeader = renderer.create(<DialogHeader />).toJSON();
        expect(dialogHeader).toMatchSnapshot();
    });

    it('mounts with the right classes', () => {
        expect(shallow(<DialogHeader />).hasClass('brew-Dialog__header')).toBe(true);
    });

    it('passes through class names correctly', () => {
        const testClassName = `class-${Math.floor(Math.random() * 1000).toString()}`;
        expect(shallow(<DialogHeader className={testClassName} />).hasClass(testClassName)).toBe(true);
    });

    it('accepts a children prop correctly', () => {
        const children = <span>children</span>;
        expect(shallow(<DialogHeader>{children}</DialogHeader>).contains(children)).toBe(true);
    });
});
