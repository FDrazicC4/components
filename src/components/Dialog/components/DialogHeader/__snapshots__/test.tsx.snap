// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogHeader matches the current snapshot 1`] = `
<div
  className="brew-Dialog__header"
/>
`;
